plugins/wasm-cpp/extensions/jwt_auth/README_EN.md

```diff
@@ -0,0 +1,399 @@
+aud: The audience(s) of the token, which is an array of case-sensitive strings.
+exp: The expiration time of the token, which is a Unix timestamp in seconds, and any token that is issued after that time is invalid.
+iat: The time at which the token was issued, also a Unix timestamp in seconds.
+jti: The unique identifier for the token, and the value is unique for every token created by the issuer. It is usually a cryptographically random value to prevent conflicts. This component adds a random entropy that an attacker cannot obtain to the structured token, making it more difficult for the token to be guessed or replayed.
+```
+
+Custom fields for a user system can also be added, such as the example below adding a "name" field for the user's nickname:
+
+```js
+{
+  "sub": "1234567890",
+  "name": "John Doe"
+}
+```
+
+The payload is then Base64-encoded, forming the second part of the JWT:
+
+`JTdCJTBBJTIwJTIwJTIyc3ViJTIyJTNBJTIwJTIyMTIzNDU2Nzg5MCUyMiUyQyUwQSUyMCUyMCUyMm5hbWUlMjIlM0ElMjAlMjJKb2huJTIwRG9lJTIyJTBBJTdE`
+
+**Signature**
+
+This part is a string that consists of the Base64-encoded header and Base64-encoded payload concatenated with a period, followed by the encryption of the resulting string using the algorithm specified in the header (where $secret represents the user's private key).
+
+```js
+var encodedString = base64UrlEncode(header) + '.' + base64UrlEncode(payload);
+var signature = HMACSHA256(encodedString, '$secret');
+```
+
+These three parts are then concatenated using periods to form the complete JWT string as shown in the example in section 1.3.2.
+
+#### 1.3.3 Time validity
+
+The gateway will verify the exp field in the token. Once this field has expired, the gateway will consider the token invalid and reject the request directly. The expiration time value must be set.
+
+1.3.4 Several Characteristics of JWT
+
+1. By default, JWT is not encrypted and cannot write secret data into JWT.
+2. JWT can not only be used for authentication but also for exchanging information. Using JWT effectively can reduce the number of times servers query the database.
+3. The biggest drawback of JWT is that the server cannot invalidate a token during use, or change the token's permission, because the server does not keep the session state. That is, once JWT is issued, it will always be valid before it expires, unless the server deploys additional logic.
+4. JWT contains authentication information itself. Once leaked, anyone can obtain all permissions of the token. To reduce theft, the expiration time of JWT should be set relatively short. For some more important permissions, users should be authenticated again when using them.
+5. To reduce theft, JWT should not be transmitted in plaintext using the HTTP protocol, and the HTTPS protocol should be used for transmission.
```

Comment:
For list items like these, should we end them with a stop or semicolon?

Reply:
It is generally recommended to end each sentence with a period for consistency and clarity when using bullet points.